server/plugins/builder/pages/index.js

```diff
@@ -25,33 +25,39 @@ class Page extends EngineBasePage {
         onPreHandler: {
           method: async (request, h) => {
             let files = fileStreamsFromPayload(request.payload)
```

Comment:
This could filter out the empty files before returning, unless we do need to delete them from `payload` below.